x/internal/outline/outline_packet_listener.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outline
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/transport/shadowsocks"
+)
+
+func NewOutlinePacketListener(accessKey string) (transport.PacketListener, error) {
```

Comment:
Argh, I just realized that there's a semantic issue here, uncovered by your change.

Is an access key the config for a VPN service, or just the transport for a tun2socks-based VPN?

I think it makes more sense to treat it as a config for the VPN service, so we can augment it with things like, the dns resolvers, the IP address to use, the routes to include/exclude....

However, we've been treating it as the transport for a tun2socks-based VPN. Kind of, since it also includes the service name.

We can pick one or the other, or say it's a VPN config that just includes the transport, but we can't say it's both.

What would happen if an accesskey encoded a wireguard config, for instance?

I'm not sure how to disentangle that right now. It needs some more thought and design.

Thoughts?

Reply:
If you want something just to use in your cli, perhaps just define the `NewOutlineClientDevice(accessKey string)` for now?

Reply:
I'm not sure if it helps, but I'll point out that the server is configured in a completely different way. That's transport only, and it uses a different format and there's no prefix or host.

Reply:
It depends, because a VPN config would also contain something that is not related to `OutlineClientDevice`, such as the TUN device name, routing table configuration, etc. But `OutlineClientDevice` may also need something more than a key, for example, the DNS resolver for connectivity test. I feel we may introduce both an `OutlineTransportOptions` `NewOutlineClientDevice(options OutlineTransportOptions)` and an `OutlineVPNOptions`:

```
type OutlineTransportOptions struct {
  OutlineServerAccessKey string
  ConnectivityTestDNSResolver string
  ...
}

type OutlineVPNOptions struct {
  Transport *OutlineTransportOptions
  Routing *RoutingTableOptions
  ...
}
```

Reply:
For now, let's just rename `accessKey` to `transportConfig`, so it's clear it must be a transport config, in case we extend the access key later.

The assumption is that the existing accessKey format is a transport config, but that can change.